packages/domain/src/use-cases/lifecycle/ISessionPresenter.ts

```diff
@@ -0,0 +1,42 @@
+import { ProfileId } from '../../entities';
+import type { LogoutReason } from '../wallets';
+
+/**
+ * The identifier of a Profile
+ */
+export type ProfileIdentifier = {
```

Comment:
Any specific reason to not use `Profile` entity?

Reply:
Passing a DTO to the presenter, not the entity directly. Leveraging TS structural typing to not create a new object until we need to.